src/main/java/wooteco/subway/domain/fare/FareByDistance.java

```diff
@@ -0,0 +1,35 @@
+package wooteco.subway.domain.fare;
+
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public enum FareByDistance {
+    BASIC(distance -> distance <= 10, distance -> 1250),
+    TENTOFIFTY(distance -> distance >= 11 && distance <= 50, distance -> 1250 + Math.max(calculateOverFromTenToFifty(distance - 10), 0)),
+    OVERFIFTY(distance -> distance >= 51, distance -> 2050 + calculateOverFifty(distance - 50));
```

Comment:
```suggestion
    OVERFIFTY(distance -> distance >= 51, distance -> 2050 + calculateOverFifty(distance));
```
형태로 변경해볼까요? 읽기가 조금 어렵네요.

calculateOverFifty() 비즈니스에 캡슐화하면 좋을 것 같아요.


예를들어 아래처럼요!

```java
private static int calculateOverFifty(double distance) {
        var overDistance = distance - 50
        return (int) ((Math.ceil(overDistance / 8) * 100));
    }
```

TENTOFIFTY 요금도 개선해보면 좋을 것 같아요.

Reply:
로직이 밖에 있으니까 가독성이 떨어지네요! 말씀해주신대로 코드를 수정했습니다 :)

Reply:
혹시 타입추론을 쓰시는 이유가 무엇인지 알 수 있을까요? 제가 타입추론에 익숙하지 않아서일지는 모르겠지만, 전 데이터타입을 명시하는게 좀 더 가독성이 좋다고 생각해서요!

Reply:
~~빠르게 설명드리려고 쓰긴했습니다~~
가장 큰 이유는 타입이 가독성이 떨어지는 것 같아서입니다! 사용하다보면 클래스 이름도 길어지고 있었는데요. IDE에서 코드추적을 잘 해줘서 큰 어려움은 없는 것 같네요 🙂